tests/jms-tests/src/test/java/org/apache/activemq/artemis/jms/tests/TemporaryDestinationTest.java

```diff
@@ -265,6 +266,126 @@ public void testTemporaryQueueDeleted() throws Exception {
       }
    }
 
+   @Test
+   public void testTemporaryQueueDeletedAfterSessionClosed() throws Exception {
```

Comment:
Can the behaviour be checked with amqp and openwire to check impl behaviour is the same

Reply:
For AMQP the handling of dynamic nodes is tested here:
https://github.com/apache/activemq-artemis/blob/master/tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/AmqpTempDestinationTest.java
And here:
https://github.com/apache/activemq-artemis/blob/master/tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSTemporaryDestinationTest.java

For OpenWire you'd want to look for a test of the RemoveInfo command that carries a DestintionInfo object.

Other tests that are specifically targeted the JMS spec (or other) compliant behaviour of a client really belong to the project that maintains the client in question.

Reply:
The issue here is not with the broker but with the core client. Therefore, I'm not sure it makes sense to test other clients on this particular use-case.